### DISCUSSION TITLE
Cannot get Multi-Output Device to work

### DISCUSSION THREAD

**Discussion by michielso** [ExistentialAudio/BlackHole#349] [Q&A]
I have been trying to get BlackHole (16ch) to work with the Multi-Output Device for months now. I have read everything I could find and have been fidling with settings and configurations. I have even bought a new mac (not related ;)) but still the sounds stops as soon as I select BlackHole 16ch.
_My configuration:_ 
A Mac Mini M1 with an external DAC (Nuforce uDac 2). Before I had a Macbook Pro from 2015 where I experienced the  same problem.

<img width="455" alt="Screenshot 2021-03-25 at 13 40 06" src="https://user-images.githubusercontent.com/30903011/112474218-3bd69a80-8d67-11eb-9d16-435873262434.png">
I can check all the devices in any configuration or order and they all work. But as soon as I select BlackHole 16ch all audio stops. If I only select the BlackHole 16ch device instead of the Multi-Output Device and try to record it will work. I have tried unplugging devices as well (expect my monitor).

Is this something for which I can start an issue? Or am I missing something else perhaps?

**Comment by devinroth** [comment]
Show a picture of when you have BlackHole with it.

Devin Roth

**Reply by michielso** [reply]
Here it is:
<img width="769" alt="Screenshot 2021-03-25 at 17 05 44" src="https://user-images.githubusercontent.com/30903011/112504724-f96f8680-8d83-11eb-8305-d5aa499949b2.png">

One weird thing happened: For the first time _ever_ it worked for a short while! After I went to the AudioMIDI Setup to make the screenshot as requested the Multi-Device was disable because the output was offline (My Bose NC Headphones, I had them turned off. Bought them after I already encountered above issue). I deselected the offline source and added my NuForce DAC and BlackHole, and they worked (a really small chance I did not select the Multi-Output Device for the sound output, but I am 90% sure I had). But now they don't. They screenshot is from when they didn't work. iTunes is frozen in the middle of a song while it gives off the appearance of playing it at this moment

I recorded some different microphones in Garageband as a test before, and made an Aggregate Device for it. Just telling everything I have done hoping it helps.

**Comment by devinroth** [comment]
It’s a known issue with multi-outputs. It’s well documented in numerous locations. Basically multi-outputs don’t always play well with devices with more than 2 channels. Try just using the 2ch version.

Devin Roth

**Reply by michielso** [reply]
Thankyou for the reply!
I installed the 2ch version after I removed the 16 ch version and it picked it up automatically. Deselecting and reselecting BlackHole resulted in the same issue as with the ch16 version though :| :
https://user-images.githubusercontent.com/30903011/112833105-caa82780-9085-11eb-81c6-c7802aa8e890.mov
(I made a screen recording with BlackHole 2ch as the audio/microphone source)

**Reply by michielso** [reply]
And to make it even more clear I made a movie from the whole process. 
https://user-images.githubusercontent.com/30903011/112834348-67b79000-9087-11eb-8deb-d89d7fc2a89f.mp4
https://user-images.githubusercontent.com/30903011/112834351-68502680-9087-11eb-8a58-1ebca10ababe.mp4
The screen recording stopped when BlackHole was being reinstalled. But I only pressed record again and did not touch anything else in the cut between the two movies.

**Comment by devinroth** [comment]
Of course it did.

Installing BlackHole restarts the whole audio system.

Devin Roth

**Reply by michielso** [reply]
I understand. I just meant to explain why it were two movies instead of one. Bottom line is there is no difference between 2ch and 16ch on my end. Although I did find a way to let it work for a moment by reinstalling every time I want to use BlackHole, which I showed in the video.

**Comment by michielso** [comment]
So back to the question at hand. Should I create an issue for this? Both BlackHole versions (2ch & 16ch) have issues operating in a Multi Output Device.

**Comment by devinroth** [comment]
Yeah. I don’t think this is related to other issues with multi outputs.

Devin Roth
